src/Compilers/Core/Portable/CommandLine/AnalyzerConfig.cs

```diff
@@ -75,19 +76,32 @@ public sealed partial class AnalyzerConfig
 
         internal ImmutableArray<Section> NamedSections { get; }
 
+        /// <summary>
+        /// Gets the concatenation of the <see cref="GlobalSection" /> and <see cref="NamedSections" />
+        /// </summary>
+        internal IEnumerable<Section> AllSections { get; }
+
         /// <summary>
         /// Gets whether this editorconfig is a topmost editorconfig.
         /// </summary>
         internal bool IsRoot => GlobalSection.Properties.TryGetValue("root", out string val) && val == "true";
 
-        private AnalyzerConfig(
+        /// <summary>
+        /// Gets whether this editorconfig is a merged global editorconfig
+        /// </summary>
+        internal bool IsMergedGlobal { get; }
+
+        internal AnalyzerConfig(
             Section globalSection,
             ImmutableArray<Section> namedSections,
-            string pathToFile)
+            string pathToFile,
+            bool isMergedGlobal)
         {
             GlobalSection = globalSection;
             NamedSections = namedSections;
+            AllSections = NamedSections.Prepend(globalSection);
```

Comment:
This boxes. The most efficient way is to give up on an abstraction and just require the utilities to access the global section and named sections separately.

Reply:
I think by making global configs a different type we don't need it anyway

---
In reply to: [421833529](https://github.com/dotnet/roslyn/pull/43889#discussion_r421833529) [](ancestors = 421833529)